### PR TITLE
enable users to specify the environments they what to track.

### PR DESCRIPTION
**Pull request by timini**
Specify in which environments you want tracking to be anabled by adding an array:
```
    metricsAdapters: [
      {
        name: 'GoogleAnalytics',
        config: {
          id: 'UA-54495053-2'
        },
        environments: ['production']
      },
      {
        name: 'Calq',
        config: {
          id: 'ca78eed5d34a041ab5cf164295cf2c25'
        },
        environments: ['production']
      },
    ]
```

[Review on Reviewable](https://reviewable.io/reviews/poteto/ember-metrics/34)

### CODE REVIEW

**Comment by poteto** [comment]
Thanks! You'll need to rebase, and fix the failing tests.

**Comment by timini** [comment]
Sorry, new to this Travis-ci thingy. Can you tell me where I find the error?

**Comment by elwayman02** [comment]
@timini click on the red x next to your latest commit to see the build errors

**Comment by timini** [comment]
@elwayman02 Tests no longer failing.

**Comment by timini** [comment]
@poteto Why do I have to rebase?

**Comment by michaeldupuisjr** [comment]
@timini It looks like the commits on this branch conflict with `master`. Rebasing with `master` gives you the opportunity to resolves these conflicts.

File: bower.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ember-metrics",
   "dependencies": {
-    "ember": "1.13.7",
+    "ember": "components/ember#release",
```
Comment (poteto) [inline bower.json]: Please revert the changes to `bower.json`, you shouldn't need to modify it.

Comment (michaeldupuisjr) [inline bower.json]: @timini  In the future, major releases of Ember will likely be incompatible with this library. We're using [Semantic Versioning](http://semver.org/), so we want to watch a specific version of Ember.

**Comment by elwayman02** [comment]
:+1: Generally you always want to rebase a PR before merging it so that you can make sure all conflicts are resolved and all tests pass before bringing the code into master. This is good practice even if there are not merge conflicts, as there may be other unintended side effects to catch.

**Comment by timini** [comment]
What should I rebase against?

**Comment by poteto** [comment]
You'll need to rebase against my `develop` branch. You can do:

```sh
$ git remote add upstream git@github.com:poteto/ember-metrics.git
$ git pull --all
$ git rebase upstream/master
```

Also, I would like to change how this is implemented. Instead of having conditionals in the `invoke` method, can you update `activateAdapters` so that we filter out adapters that don't have environments that match the current environment?

E.g.

```js
activateAdapters(adapterOptions = []) {
  const appEnv = get(this, 'environment');
  const cachedAdapters = get(this, '_adapters');
  const activatedAdapters = {};

  adapterOptions.filter((adapter) => adapter.environments.contains(appEnv))
    .forEach((adapterOption) => {
      const { name } = adapterOption;
      const adapter = cachedAdapters[name] ? cachedAdapters[name] : this._activateAdapter(adapterOption);

      set(activatedAdapters, name, adapter);
    });

  return set(this, '_adapters', activatedAdapters);
}
```

You should also add a test for this. 

Thanks!

File: bower.json [hunk 2]
```diff
@@ -12,5 +12,8 @@
     "loader.js": "ember-cli/loader.js#3.2.1",
     "qunit": "~1.18.0",
     "sinonjs": "~1.14.1"
+  },
+  "resolutions": {
```
Comment (poteto) [inline bower.json]: Remove this resolution, please!

**Comment by elwayman02** [comment]
:+1: to @poteto's idea of filtering adapters against the env

**Comment by sly7-7** [comment]
@poteto I may have missed something but I think it could be closed :smile:
